Update acid test

vladimyr opened the pull request:
<!--
Thank you for your pull request. Please review the below requirements.

Contributor guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change
<!-- Provide a description of the change -->
Joined two-step acid test into a single c/p-able shell snippet.

cclauss commented on macOS_Catalina.md:
````diff
@@ -17,14 +17,28 @@ If `ProductVersion` is less then `10.15` then this document is not for you. Norm
-If both tests succeeded, _you are done_! You should be ready to install `node-gyp`.
+```
+/usr/sbin/pkgutil --pkgs |
+  grep -i com.apple.pkg.cltools |
````
https://github.com/nodejs/node-gyp/blob/972780bde72b9a61153f3ea7ea9700cec001ffd7/gyp/pylib/gyp/xcode_emulation.py#L1436-L1438

vladimyr replied:
Which would indicate that current docs are wrong too? 😕

vladimyr replied:
@cclauss How about using this snippet:
```sh
#!/bin/bash

pkgs=(
  "com.apple.pkg.DeveloperToolsCLILeo" # standalone
  "com.apple.pkg.DeveloperToolsCLI"    # from XCode
  "com.apple.pkg.CLTools_Executables"  # Mavericks
)

for pkg in "${pkgs[@]}"; do
  output=$(/usr/sbin/pkgutil --pkg-info "$pkg" 2>/dev/null)
  if [ "$output" ]; then
    version=$(echo "$output" | grep 'version' | cut -d' ' -f2)
    break
  fi
done

if [ "$version" ]; then
  echo "Command Line Tools version: $version"
else
  echo >&2 'Command Line Tools not found'
fi
```

vladimyr commented:
I've converted the acid test to a remote shell script and introduced custom `git.io` URL mapping:
```
curl https://git.io/ -i -F "url=https://raw.githubusercontent.com/nodejs/node-gyp/master/acid-test.sh" -F "code=node-gyp:acid-test"
```
As a result, end-users can simply invoke it using:
```
curl -L https://git.io/node-gyp:acid-test
```

rvagg commented on macOS_Catalina.md:
````diff
@@ -17,14 +17,13 @@ If `ProductVersion` is less then `10.15` then this document is not for you. Norm
-    
-If both tests succeeded, _you are done_! You should be ready to install `node-gyp`.
+```
+curl -L https://git.io/node-gyp:acid-test | bash
````
I'm not entirely comfortable promoting a shortened link, piping anything to bash involves a lot of trust so I'd rather this be explicit _and_ also present a more manual way of doing it. You could update your text to point directly to `acid-test.sh` if they want to download it themselves, and then use the full URL as an option if they're comfortable: `https://github.com/nodejs/node-gyp/blob/master/acid-test.sh`

_But_ also this test needs to be renamed, I know this makes it tedious, but this is only an issue with macOS Catalina and we need to separate this out as a concern. I suggest: `macOS_Catalina_acid_test.sh` so it shows up right next to this doc.

I'm not vouching for the accuracy here btw. That's another matter which needs signoff.

vladimyr replied:
> I'm not entirely comfortable promoting a shortened link, piping anything to bash involves a lot of trust so I'd rather this be explicit _and_ also present a more manual way of doing it. You could update your text to point directly to `acid-test.sh` if they want to download it themselves, and then use the full URL as an option if they're comfortable: `https://github.com/nodejs/node-gyp/blob/master/acid-test.sh`

Understood that and will expand it back to the full URL although it needs to be /raw instead of /blob.

> _But_ also this test needs to be renamed, I know this makes it tedious, but this is only an issue with macOS Catalina and we need to separate this out as a concern. I suggest: `macOS_Catalina_acid_test.sh` so it shows up right next to this doc.

Sure, it perfectly makes sense 👍

cclauss commented:
Homebrew now kicks out the following message:
```
Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you an update run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.
```
This worked like a champ for me.  I did the first command and about 30 seconds later a macOS __System Preferences/Software Update__ dialog box popped up asking me to perform and update.
```
com.apple.pkg.CLTools_Executables
com.apple.pkg.CLTools_SDK_macOS1015
com.apple.pkg.CLTools_SDK_macOS1014
com.apple.pkg.CLTools_macOS_SDK
```
Should we adopt this approach?

vladimyr commented:
> [Homebrew](https://brew.sh/) now kicks out the following message:
> ```
> Warning: A newer Command Line Tools release is available.
> Update them from Software Update in System Preferences or run:
>   softwareupdate --all --install --force
> 
> If that doesn't show you an update run:
>   sudo rm -rf /Library/Developer/CommandLineTools
>   sudo xcode-select --install
> 
> Alternatively, manually download them from:
>   https://developer.apple.com/download/more/.
> ```
> This worked like a champ for me.  I did the first command and about 30 seconds later a macOS __System Preferences/Software Update__ dialog box popped up asking me to perform an update.

@cclauss It happened to me too and I did the same except it totally corrupted my clt installation and the only way out was to remove it completely and manually install again by downloading from Apple's web (last resort described here). Simply put it isn't safe!

rvagg commented:
f7bfce96edd

vladimyr commented:
@rvagg Be warned that this isn't reliable at all. All of a sudden it gives me false negative on my own machine!

I essentially rewrote Python code that @cclauss pointed to in bash but that is yesterday news. Today's HB CLT detection looks much more sophisticated https://github.com/Homebrew/brew/blob/5fb158f/Library/Homebrew/os/mac/xcode.rb#L198-L326
